### PR TITLE
Re-introduce beta features

### PR DESCRIPTION
**Pull request by prprabhu-ms**
* Re-introduce beta features removed for GA
* Introduce conditional compilation, building towards a `stable` flavor build.

These are pulled from the `feature/conditional-compilation` branch.

### CODE REVIEW

File: change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```
Comment (prprabhu-ms) [inline change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json:2]: Should this be a patch change?

Comment (JamesBurnside) [reply change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json:2]: maybe we keep prelease as an option?

File: packages/chat-stateful-client/src/ChatContext.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ enableMapSet();
  */
 export class ChatContext {
   private _state: ChatClientState = {
-    userId: <UnknownIdentifierKind>{ id: '' },
```
Comment (prprabhu-ms) [inline packages/chat-stateful-client/src/ChatContext.ts:30]: Should this be removed?

File: packages/react-components/src/components/MessageThread.tsx [hunk 3]
```diff
@@ -244,7 +244,7 @@ const ParticipantSystemMessageComponent = ({
   if (participantsStr !== '') {
     return (
       <SystemMessageComponent
-        iconName={(message.iconName ?? '') as SystemMessageIconTypes}
+        iconName={(message.iconName ? message.iconName : '') as SystemMessageIconTypes}
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/MessageThread.tsx:247]: Check this is intended.

File: packages/storybook/stories/QuickStarts/QuickstartComposite.stories.mdx [hunk 4]
```diff
@@ -92,6 +92,25 @@ The following classes and interfaces handle some of the major features of the Az
 | [CallComposite](./?path=/story/composites-call-basicexample--basic-example) | Composite component that renders a calling experience with participant gallery and controls. |
 | [ChatComposite](./?path=/story/composites-chat-basicexample--basic-example) | Composite component that renders a chat experience with chat thread and input                |
 
+<<<<<<< HEAD
```
Comment (prprabhu-ms) [inline packages/storybook/stories/QuickStarts/QuickstartComposite.stories.mdx:95]: FIXME

File: change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json (hunk 1 above)
Comment (prprabhu-ms) [reply change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json:2]: My comment was more about how to label storybook changes. They don't impact the library at all. So... none? Or patch?

Comment (alcail) [reply change/@internal-storybook-dc3f30cf-69d5-4778-9f40-fb10fae10863.json:2]: if no impact, I would say "none"

File: samples/Chat/src/app/ChatHeader.tsx [hunk 5]
```diff
@@ -26,6 +30,14 @@ export const ChatHeader = (props: ChatHeaderProps): JSX.Element => {
 
   return (
     <Stack horizontal={true} verticalAlign={'center'} horizontalAlign="end" className={chatHeaderContainerStyle}>
+      {/** FIXME: Similar to #1235 */}
+      <div className={paneButtonContainerStyle}>
+        <IconButton
+          onRenderIcon={() => (props.isParticipantsDisplayed ? <People20Filled /> : <People20Regular />)}
+          className={mergeStyles({ color: theme.palette.neutralPrimaryAlt })}
+          onClick={() => props.setHideParticipants(props.isParticipantsDisplayed)}
```
Comment (alcail) [inline samples/Chat/src/app/ChatHeader.tsx:38]: missing ariaLabel prop on this one. As it's an icon with no label, screen reader will not know what it is apart being a button, so we need to add an ariaLabel string like ' Participant list' or 'participant pane'

**Review by alcail — APPROVED** [review]

File: samples/Chat/src/app/ChatHeader.tsx (hunk 5 above)
Comment (prprabhu-ms) [reply samples/Chat/src/app/ChatHeader.tsx:38]: I've added a FIXME above, will upload a follow up PR. I don't want to make new functional changes in this PR. This is as much as possible:

`git diff --binary main..<feature-branch-sha> | git patch` :)

File: change/@internal-react-components-f0a9c548-cc05-43c2-a29d-30b027714698.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "comment": "Conditional build of stable flavor",
```
Comment (JamesBurnside) [inline change/@internal-react-components-f0a9c548-cc05-43c2-a29d-30b027714698.json:2]: not blocking this PR, but we need all this documented well in https://github.com/Azure/communication-ui-library/tree/main/docs/infrastructure

Comment (prprabhu-ms) [reply change/@internal-react-components-f0a9c548-cc05-43c2-a29d-30b027714698.json:2]: Oooh yes. We need that. We're not fully done here actually. @PorterNan let's together aim to document all this by the end of this sprint.

File: samples/StaticHtmlComposites/playwright.config.ts [hunk 7]
```diff
@@ -1,24 +0,0 @@
-// playwright.config.ts
```
Comment (JamesBurnside) [inline samples/StaticHtmlComposites/playwright.config.ts:1]: intentionally deleted?

File: packages/storybook/stories/Adapters/Adapters.stories.mdx [hunk 8]
```diff
@@ -32,8 +31,6 @@ Teams Meeting information can be acquired through [Microsoft Graph](https://docs
 
 <Source code={CallAdapterExampleText} />
 
-<MessageBar>Note: The `displayName` in Call Composite can be a maximum of 256 characters.</MessageBar>
```
Comment (JamesBurnside) [inline packages/storybook/stories/Adapters/Adapters.stories.mdx:35]: Are these changes in this file intentional?

File: packages/react-composites/tests/browser/chat/ChatComposite.test.ts [hunk 9]
```diff
@@ -106,11 +112,14 @@ test.describe('Chat Composite custom data model', () => {
   });
 
   test('can be viewed by user[1]', async ({ pages }) => {
-    const testMessageText = 'How the turn tables';
     const page = pages[0];
     await page.bringToFront();
-    await sendMessage(page, testMessageText);
-    await waitForMessageDelivered(page);
+    await page.type(dataUiId(IDS.sendboxTextField), 'How the turn tables');
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/chat/ChatComposite.test.ts:117]: these changes look like needing reverted

File: packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx [hunk 10]
```diff
@@ -55,7 +55,12 @@ export const ContosoChatContainer = (props: ContainerProps): JSX.Element => {
   return (
     <div style={{ height: '100vh', width: '100vw' }}>
       {adapter ? (
-        <ChatComposite fluentTheme={props.fluentTheme} locale={props.locale} adapter={adapter} />
+        <ChatComposite
+          fluentTheme={props.fluentTheme}
+          locale={props.locale}
+          adapter={adapter}
+          options={{ participantPane: true }}
```
Comment (JamesBurnside) [inline packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx:62]: This presumably can be in the PR that sets up CI for stable build, but I think everywhere with the participantPane property needs `/* @conditional-compile-remove-from(stable) */`

File: samples/StaticHtmlComposites/playwright.config.ts (hunk 7 above)
Comment (prprabhu-ms) [reply samples/StaticHtmlComposites/playwright.config.ts:1]: Yes: https://github.com/Azure/communication-ui-library/pull/1252/files?authenticity_token=8apjI8uIvIx6QwC7W4PKz%2F8mXyx9YQ08zWJROjnC8Ss1bZpWYAul2KyJH62blWMX6urx1xHdKS1nUGU0nQWTEQ%3D%3D&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.ts&file-filters%5B%5D=.yaml#diff-a7e4059ab620c34edf008f471b5543bfdd41b0da2f2c663c7a327485b1048818L1

File: packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx (hunk 10 above)
Comment (prprabhu-ms) [reply packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx:62]: Yeah, the conditional compilation work isn't complete by any means. But the beta build is unaffected, and is what our CI tests anyway. So deduping first.

Comment (prprabhu-ms) [reply packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx:62]: @PorterNan  ^^

File: packages/storybook/stories/Adapters/Adapters.stories.mdx (hunk 8 above)
Comment (prprabhu-ms) [reply packages/storybook/stories/Adapters/Adapters.stories.mdx:35]: Thanks, good catch.

File: packages/react-composites/tests/browser/chat/ChatComposite.test.ts (hunk 9 above)
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/chat/ChatComposite.test.ts:117]: Actually, only partially. I need to merge.

File: common/config/rush/command-line.json [hunk 11]
```diff
@@ -31,6 +31,25 @@
       "shellCommand": "node common/scripts/ensure-consistent-versions.js",
       "safeForSimultaneousRushProcesses": true
     },
+    {
+      "commandKind": "bulk",
+      "name": "build:stable",
+      "summary": "Preprocess and build the GA version.",
```
Comment (JamesBurnside) [inline common/config/rush/command-line.json:37]: nit: consistency with summarys; this says "build GA version" but below we say for "non-beta version".
```suggestion
      "summary": "Preprocess and build only stable (non-beta) code in packages",
```

File: samples/StaticHtmlComposites/playwright.config.ts (hunk 7 above)
Comment (JamesBurnside) [reply samples/StaticHtmlComposites/playwright.config.ts:1]: Ah this was an obsolete file now it exists in `tests/` directory

**Review by JamesBurnside — APPROVED** [review]

File: samples/Chat/src/app/ChatHeader.tsx (hunk 5 above)
Comment (prprabhu-ms) [reply samples/Chat/src/app/ChatHeader.tsx:38]: https://github.com/Azure/communication-ui-library/pull/1261 up for this.

File: packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx (hunk 10 above)
Comment (prprabhu-ms) [reply packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx:62]: @JamesBurnside Actually there is no conditional compilation in storybook. It builds against the beta API, and we call out what features are in preview vs not.

Comment (JamesBurnside) [reply packages/storybook/stories/ChatComposite/CustomBehaviorExampleContainer.tsx:62]: Good point!

**Pull request merged**